.github/workflows/update-copyright-notices.yml

```diff
@@ -0,0 +1,46 @@
+name: Update copyright notices
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  check-and-update-copyright-notices:
+    if: ${{ github.event_name == 'push' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./tools/update-copyright-notices.sh
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "github-actions[bot]"
+          pr_branch=updateCopyrightNotices
+          git checkout -b ${pr_branch}
+          git add .
+          git commit -m "AUTO: Update copyright notices for $(date +%Y)" || exit 0
+          git push origin "+${pr_branch}"
+          existing_pr=$(gh pr list --head "${pr_branch}" --json number --jq '.[].number')
+          if [[ $existing_pr ]]; then
+            gh pr comment "${existing_pr}" --body "PR has been updated again."
```

Comment:
@gilgongo not sure if this message is clear enough?

Reply:
What should be changes/added?

Reply:
We should clarify *why* the PR was updated. Is it possible to link the related commit? Then something like:

Updated this PR because of commit xyz
At least for me it wasn't quite clear why this message would come up.

Reply:
> We should clarify why the PR was updated.

Is it really relevant? The updated diff is easily visible from the PR. We don't explain the *why* in more depth when initially creating the PR either. The answer is simple: The script becomes active (opens a PR / updates it) when there's something to fix. This might be due to the year rolling over or due to some old copyright header being introduced.

Looking up the commit which triggered is probably possible in a simple way. But remember that this is push-based, so it will not be easy to tell if it's this commit or rather some parent.

To sum it up: It would certainly be possible, but it would lead to way more code/complexity.

Reply:
Ok. So you'd say it's not worth it. I'm still not happy with a message saying (Yeah, this PR just got updated). What speaks against omitting it completely? What speaks for having it? I mean, you'll always see on GH that the PR got force-pushed and hence updated.

Reply:
Well, it's an edge case (IMO).
I was mostly thinking about protecting against unwanted merges.
E.g.:
- PR is reviewed and close to being merged
- Script updates PR
- PR is merged, thinking that it had been reviewed, without anyone noticing that it was changed meanwhile

This can happen with human PRs as well, but humans might act in good faith and provide a pointer, so I figured the script should try do the same. :)

Reply:
In which case 
> "PR has been updated again."

could be
> "PR was updated by tools/update-copyright-headers"

or similar, to make it clear.

Reply:
Done.

Reply:
Ok. Probably it’s ok then.